src/index.ts

```diff
@@ -156,24 +161,34 @@ export class Spring {
     // being changed in `updatedConfig`, we run the simulation with `_step()`
     // and default `fromValue` and `initialVelocity` to their current values.
 
-    this._advanceSpringToTime(Date.now());
+    this._advanceSpringToTime();
 
-    const baseConfig = {
-      fromValue: this._currentValue,
-      initialVelocity: this._currentVelocity
-    };
+    this._config.fromValue = this._currentValue;
+    this._config.initialVelocity = this._currentVelocity;
 
-    this._config = {
-      ...this._config,
-      ...baseConfig,
-      ...updatedConfig
-    };
+    for (const key in updatedConfig) {
+      if (this._config.hasOwnProperty(key)) {
+        this._config[key] = updatedConfig[key as keyof SpringConfig];
+      }
+    }
 
     this._reset();
 
     return this;
   }
 
+  /**
+   *
+   */
+  setValue(value: number): this {
+    this._config.fromValue = this._currentValue;
+    this._config.toValue = value;
+    this._config.initialVelocity = this._currentVelocity;
+    this.start();
+    this._advanceSpringToTime(undefined, true);
```

Comment:
I definitely don't like this `(undefined, true)`. Perhaps it'd be best to move the raf loop into start/stop and the spring logic in step?